test/run-tests.sh

```diff
@@ -1,10 +1,41 @@
 #!/usr/bin/env bash
+set -e  # stop the script when any command fails
 
-# Return failure as soon as a command fails to execute
+cd "$(dirname "$0")"
 
-set -e
+# Parse arguments and provide usage message.
+# Parse options in pure Bash because macOS `getopt` is stuck
+# in what util-linux `getopt` calls `GETOPT_COMPATIBLE` mode.
 
-cd "$(dirname "$0")"
+usage() {
+	echo "Runs regression tests on RGBDS."
+	echo "Options:"
+	echo "    -h, --help      show this help message"
+	echo "    --only-free     skip tests that build nonfree codebases"
+}
+
+WITH_NONFREE=true
+FETCH_CMD="fetch-test-deps.sh"
```

Comment:
Nitpick: other scripts generally use lowercase variables, except for the names of programs, which are then named after themselves not <code><i>NAME</i>_CMD=./<i>NAME</i></code>.
```suggestion
nonfree=true

FETCH_TEST_DEPS="fetch-test-deps.sh"
```

Reply:
Why was `with_` also removed?

Reply:
Just read the existing scripts. I called it `nonfree` not `with_nonfree` because it's shorter and equally explanatory.

Reply:
I don't trust myself to learn anything actionable through "Just read[ing] the existing scripts" because the last time I "Just read the existing scripts", I ended up keeping a [comment in the existing `run-tests.sh`](https://github.com/gbdev/rgbds/blob/168950855d4d06eb4354e781b20da65a6833763b/test/run-tests.sh#L3C1-L3C55) that you told me was better off removed.

Reply:
Apologies for that. We don't have an official style guide, and my own review preferences aren't going to be consistent or applied everywhere. Feel free to push back if I sound nitpicky, or get ISSOtm's opinion too.